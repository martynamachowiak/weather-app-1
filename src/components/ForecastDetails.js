import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const ForecastDetails = ({ forecast }) => {
  const { date, humidity, temperature, wind } = forecast;
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__humidty">{humidity}</div>
      <div className="forecast-details__temperature">
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__wind__speed">{wind.speed}</div>
      <div className="forecast-details__wind__direction">{wind.direction}</div>
    </div>
  );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
