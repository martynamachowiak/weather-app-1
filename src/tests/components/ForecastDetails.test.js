import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
    const validProps = {
      date: 11111111,
      humidity: 100,
      temperature: {
        min: 12,
        max: 22,
      },
      wind: {
        speed: 100,
        direction: "nw",
      },
    };
  
    it("renders correctly", () => {
        const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});