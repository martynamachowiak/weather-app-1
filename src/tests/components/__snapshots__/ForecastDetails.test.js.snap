// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-details__humidty"
    >
      100
    </div>
    <div
      class="forecast-details__temperature"
    >
      12°C
    </div>
    <div
      class="forecast-details__temperature"
    >
      22°C
    </div>
    <div
      class="forecast-details__wind__speed"
    >
      100
    </div>
    <div
      class="forecast-details__wind__direction"
    >
      nw
    </div>
  </div>
</DocumentFragment>
`;
